### DISCUSSION TITLE
Push certificate generation via API

### DISCUSSION THREAD

**Discussion by tmaly1980** [fastlane/fastlane#17951] [Q&A]
Is it possible to generate push certificates through the new AppStoreConnect API? Apple is enforcing 2FA for everyone starting in February, so it needs to be figured out somehow. I can't find any mention in fastlane or Apple's official docs. What are the alternatives if it's just not possible?

**Comment by dinsen** [comment]
Hi @tmaly1980 👋 
It dosen't look like there is an endpoint to generate push certificates. 
Have you considered using an Auth key instead? 
https://developer.clevertap.com/docs/how-to-create-an-ios-apns-auth-key
